pkg/mesh-discovery/translation/workload/detector/istio/istio_workload_detector.go

```
@@ -0,0 +1,333 @@
+		if hasMeshProxyContainer(workload) {
+			// the workload is either a gateway or has had its proxy manually injected
+			return mesh
+		}
+
+		isInjected := d.meshInjectsWorkload(
+			d.ctx,
+			workload,
+			mesh,
+		)
+
+		if isInjected {
+			return mesh
+		}
+
+	}
+	return nil
+}
+
+// returns true if the given workload contains a proxy container that has been manually added
+func hasMeshProxyContainer(
+	workload types.Workload,
+) bool {
+	// TODO(ilackarms): currently we assume one mesh per cluster,
+	// and that the control plane for a given proxy is always
+	// the mesh
+	return containsSidecarContainer(workload.GetPodTemplate().Spec.Containers)
+}
+
+// returns true if the given workload is injected by the given mesh (istiod) instance
+func (d workloadDetector) meshInjectsWorkload(
+	ctx context.Context,
+	workload types.Workload,
+	mesh *v1.Mesh,
+) bool {
+
+	namespaceInjected := d.injectedNamespaces[namespaceKey(workload.GetNamespace(), workload.GetClusterName())]
+
+	cfg, ok := d.cachedInjectorConfigs[sets.Key(mesh)]
+	if !ok {
+		var err error
+		cfg, err = getInjectorConfig(mesh.Spec.GetIstio(), d.configMaps)
+		if err != nil {
+			recordInjectionConfigParseFailed(ctx, mesh, eris.Wrap(err, "getting injector config for mesh"))
+		}
+		d.cachedInjectorConfigs[sets.Key(mesh)] = cfg
+	}
+
+	return injectRequired(
+		ctx,
+		ignoredNamespaces,
+		namespaceInjected,
+		cfg,
+		workload,
+	)
+}
+
+func getInjectorConfig(
+	istioMesh *v1.MeshSpec_Istio,
+	configMaps corev1sets.ConfigMapSet,
+) (*inject.Config, error) {
+	injectorCm, err := configMaps.Find(&skv1.ClusterObjectRef{
+		Name:        getInjectorConfigMapName(""), // TODO(ilackarms): support mesh revisions here
+		Namespace:   istioMesh.GetInstallation().GetNamespace(),
+		ClusterName: istioMesh.GetInstallation().GetCluster(),
+	})
+	if err != nil {
+		// TODO(ilackarms): no injector configured for this mesh, explore other options for workload detection
```

we'll want to return an error here so we can track occurrences of this scenario

---

i don't think it's technically an error. istio is able to start up (and will use a default config) for the sidecar injector in this case. we do the same - if no config is found, we default it in a way that mirrors istio